### PR TITLE
Update instance-segmentation tutorial documentation

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

Rewrote the Instance-segmentation tutorial, which was using the toy dataset, by modifying it to use a WGISD dataset like detection.
(car_tree_bug -> WGISD dataset)

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst [hunk 1]
```diff
@@ -43,40 +43,76 @@ environment:
...
-    ├── instances_train.json
-    └── instances_val.json
+      ├── instances_train.json
+      ├── instances_val.json
+      (Optional)
+      └── instances_test.json
   ├──images/
-    └── <images>
-  ...
+      (Optional)
```
Comment (eunwoosh) [inline docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst:92]: Is this right place for `(Optional)`?

File: docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst [hunk 2]
```diff
@@ -134,75 +178,101 @@ It will create **otx-workspace-INSTANCE_SEGMENTATION** with all necessary config
...
+                              --learning_parameters.num_iters 100 \
+                              --learning_parameters.enable_early_stopping false
 
-  ...
-  2023-02-21 22:34:53,474 | INFO : Update LrUpdaterHook patience: 5 -> 2
-  2023-02-21 22:34:53,474 | INFO : Update CheckpointHook interval: 1 -> 5
-  2023-02-21 22:34:53,474 | INFO : Update EvalHook interval: 1 -> 5
-  2023-02-21 22:34:53,474 | INFO : Update EarlyStoppingHook patience: 10 -> 3
-  2023-02-21 22:34:54,320 | INFO : Epoch [1][2/2] lr: 3.400e-04, eta: 3:14:44, time: 1.180, data_time: 0.784, memory: 7322, current_iters: 1, loss_rpn_cls: 0.0720, loss_rpn_bbox: 0.0250, loss_cls: 2.6643, acc: 89.3066, loss_bbox: 0.3984, loss_mask: 3.5540, loss: 6.7136, grad_norm: 66.2921
+5. The training results are ``weights.pth`` and ``label_schema.json`` files that located in ``outputs/**_train/models`` folder,
```
Comment (eunwoosh) [inline docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst]:
```suggestion
5. The training results are ``weights.pth`` and ``label_schema.json`` files located in ``outputs/**_train/models`` folder, 
```

File: docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst [hunk 3]
```diff
@@ -134,75 +178,101 @@ It will create **otx-workspace-INSTANCE_SEGMENTATION** with all necessary config
...
+          └── cli_report.log
   ...
-  2023-02-21 22:35:07,908 | INFO : Inference completed
-  2023-02-21 22:35:07,908 | INFO : called evaluate()
-  2023-02-21 22:35:07,909 | INFO : F-measure after evaluation: 0.33333333333333326
-  2023-02-21 22:35:07,909 | INFO : Evaluation completed
-  Performance(score: 0.33333333333333326, dashboard: (1 metric groups))
 
-After that, we have the PyTorch instance segmentation model trained with OpenVINO™ Training Extensions, which we can use for evaluation, export, optimization and deployment.
+After that, we have the PyTorch instance segmentation model trained with OpenVINO™ Training Extensions, which we can use for evaliation, export, optimization and deployment.
```
Comment (eunwoosh) [inline docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst]:
```suggestion
After that, we have the PyTorch instance segmentation model trained with OpenVINO™ Training Extensions, which we can use for evaluation, export, optimization and deployment.
```

**Review by eunwoosh — COMMENTED** [review]
Thanks for update :) I left some comments. please take a look.

File: docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst (hunk 1 above)
Comment (harimkang) [reply docs/source/guide/tutorials/base/how_to_train/instance_segmentation.rst:92]: Yes, this is not a requirement when training.

**Review by eunwoosh — DISMISSED** [review]

**Comment by sungmanc** [comment]
@goodsong81 , @harimkang , @eunwoosh  I think this PR could be applied to release/1.2.1 since it could bring more useful examples. What do you think?

**Review by sungmanc — APPROVED** [review]

**Comment by eunwoosh** [comment]
> @goodsong81 , @harimkang , @eunwoosh I think this PR could be applied to release/1.2.1 since it could bring more useful examples. What do you think?

I agreed. or just merging it to releases/1.2.1 and merge back into develop branch is also possible if it's not urgent.

**Review by sungmanc — CHANGES_REQUESTED** [review]
Need to decide the target of this PR

**Comment by goodsong81** [comment]
OK. Let's retarget to 1.2.1.

**Review by goodsong81 — DISMISSED** [review]

**Review by sungmanc — APPROVED** [review]

**Comment by harimkang** [comment]
Reopen the PR here (https://github.com/openvinotoolkit/training_extensions/pull/2082) by retargeting to releases/1.2.1.

**Pull request closed without merging**